src/MatrixClient.ts

```diff
@@ -897,6 +898,34 @@ export class MatrixClient extends EventEmitter {
         }
     }
 
+    /**
+     * This returns a list of message and state events for a room.
+     *
+     * It uses pagination query parameters to paginate history in the room.
+     *
+     * @param roomId the room ID to get the event in
+     * @param dir The direction to return events from.
+     *            If this is set to `f`, events will be returned in chronological order starting at `from`.
+     *            If it is set to `b`, events will be returned in reverse chronolgical order, again starting at `from`.
+     * @param from The token to start returning events from.
+     *             This token can be obtained from a prev_batch or next_batch token returned by the /sync endpoint,
+     *             or from an end token returned by a previous call to this function.
+     * @param filter A JSON RoomEventFilter to filter returned events with.
```

Comment:
We shouldn't be forcing the caller to encode their filter for us

Reply:
Ah yes, that would be silly.